packages/opentelemetry-node/test/testutils.js

```diff
@@ -332,6 +336,16 @@ class TestCollector {
 
             return aStartInt < bStartInt ? -1 : aStartInt > bStartInt ? 1 : 0;
         });
+
+        // NOTE: some resource detectors are starting spans and this affects
+        // the tests we have implmented until now (2024-06)
+        // - internal components like detector shouldn't be crating traces
```

Comment:
```suggestion
        // NOTE: Ignore spans from the GCP detector for testing. It shouldn't be
        // creating spans, but that should be fixed upstream.
```

Perhaps an internal issue we can track to know when this hack can be removed?

Reply:
https://github.com/elastic/elastic-otel-node/issues/222